donazioni/viste.py

```diff
@@ -0,0 +1,27 @@
+from django.http import Http404
+from django.shortcuts import redirect
+from django.utils.decorators import method_decorator
+from django.views.generic import TemplateView
+
+from anagrafica.permessi.costanti import GESTIONE_CAMPAGNE, GESTIONE_CAMPAGNA, ERRORE_PERMESSI
+from autenticazione.funzioni import pagina_privata, VistaDecorata
+
+
+class DonazioniWelcomePage(VistaDecorata, TemplateView):
```

Comment:
Come discusso, questo passerebbe la review, ma in linea generale e' uno di quei casi dove coerenza interna batte utilita'. In particolare, la gran maggioranza di viste in Gaia usa viste a funzione. Nonostante non e' necesseriamente la cosa migliore, tutte le funzioni accessorie (come i decoratori per l'autenticazione) sono sviluppate con questo in mente. 

A scopo esemplificativo, puoi sovrascrivere l'intera classe in una funzione di solo qualche riga:

```python
@pagina_privata(permessi=(GESTIONE_CAMPAGNE,))
def donazioni(request, me):
  contesto = {"sedi": me.oggetti_permesso(GESTIONE_CAMPAGNE),
              "campagne": me.oggetti_permesso(GESTIONE_CAMPAGNA)}
  return "donazioni.html", contesto
```

Nota che il parametro `permessi` al decoratore `GESTIONE_CAMPAGNE` e' qui ridondante e ignorarlo non causerebbe un problema di sicurezza. Infatti e' possibile caricare la vista, ma se l'utente identificato non ha deleghe, `me.oggetti_permesso(GESTIONE_CAMPAGNE)` non ritornerebbe niente.